### PR TITLE
[12.0][MIG] sale_triple_discount

### PR DESCRIPTION
**Pull request by tafaRU**
PR #1000 :tada: :tada: :tada:

### CODE REVIEW

**Comment by tafaRU** [comment]
@mpanarin the email address used in https://github.com/OCA/sale-workflow/pull/1000/commits/7817083b956ba5162df5812ede3abbe3031acd4f it's different as the one it's registered.
Could you please tell me the right one I can use by amending that commit?

**Review by stevech091 — APPROVED** [review]
Functional review OK

**Comment by mpanarin** [comment]
@tafaRU sorry, I missed your message
you can use mykhailopanarin**at**gmail.com

File: sale_triple_discount/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+{
+    'name': 'Sale Triple Discount',
+    'version': '12.0.1.0.0',
+    'category': 'Sales Management',
+    'author': 'ADHOC SA, '
+              'Agile Business Group, '
+              'Tecnativa, '
+              'Odoo Community Association (OCA)',
+    'website': 'https://odoo-community.org',
```
Comment (nikul-serpentcs) [inline sale_triple_discount/__manifest__.py]: @tafaRU https://github.com/OCA/sale-workflow

**Review by nikul-serpentcs — COMMENTED** [review]
minor change

**Comment by tafaRU** [comment]
@mpanarin  I used mykhailopanarinatgmail.com to amend your commit. Thanks!

File: sale_triple_discount/__manifest__.py (hunk 1 above)
Comment (tafaRU) [reply sale_triple_discount/__manifest__.py]: done, thanks!

**Comment by tafaRU** [comment]
@nikul-serpentcs could you please update your review?
And @mpanarin could you please take the opportunity to have one?

Thank you in advance.

File: sale_triple_discount/__manifest__.py [hunk 2]
```diff
@@ -1,23 +1,22 @@
...
+# Copyright 2017 - 2019 Alex Comba - Agile Business Group
 # Copyright 2017 Tecnativa - David Vidal
 # Copyright 2018 Jacques-Etienne Baudoux (BCIM sprl) <je@bcim.be>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 {
     'name': 'Sale Triple Discount',
-    'version': '10.0.1.0.1',
+    'version': '12.0.1.0.0',
     'category': 'Sales Management',
```
Comment (rousseldenis) [inline sale_triple_discount/__manifest__.py]: Please change it to 'Sales' to appear correctly in OCA shop.

**Review by rousseldenis — APPROVED** [review]
Code review. LGTM

We could enhance this one adding tests for missing parts as this module caused headaches on previous branches.

**Review by andreampiovesana — APPROVED** [review]
essential in italy

**Review by mpanarin — APPROVED** [review]

File: sale_triple_discount/__manifest__.py (hunk 2 above)
Comment (tafaRU) [reply sale_triple_discount/__manifest__.py]: Thanks @rousseldenis, changed in my last force pushed commit.

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1:

**Comment by tafaRU** [comment]
@OCA/crm-sales-marketing-maintainers could it be merged?

File: sale_triple_discount/models/sale_order_line.py [hunk 3]
```diff
@@ -136,6 +135,6 @@ def triple_discount_postprocess(self, prev_values):
         Updating the cache provides consistency through recomputations."""
         self.invalidate_cache(
             fnames=['discount', 'discount2', 'discount3'],
-            ids=[l.id for l in prev_values.keys()])
-        for line, prev_vals_dict in prev_values.items():
+            ids=[l.id for l in list(prev_values.keys())])
+        for line, prev_vals_dict in list(prev_values.items()):
```
Comment (simahawk) [inline sale_triple_discount/models/sale_order_line.py:139]: this call to `list` is not needed, non blocking

Comment (tafaRU) [reply sale_triple_discount/models/sale_order_line.py:139]: ok @simahawk, `2to3` is the responsible of this change :wink:

**Pull request merged**